### ISSUE TITLE
Date parsing does not always work with custom input formats

### ISSUE THREAD

**Issue by charliehw** [BladeRunnerJS/brjs#1270]
`br.parsing.DateParser` does some quasi-localisation in its `_standardizeDateSeparators` method. However, this prevents dates with "." separators from being parsed when supplying a custom input format.

e.g. it fails to parse "09.03.2015" as "DD.MM.YYYY" because it first converts the input value to "09-03-2015".

    var parser = new br.parsing.DateParser();

    parser.parse('09.03.2015', {
        inputFormats: 'DD.MM.YYYY',
        outputFormat: 'YYYYMMDD'
    }); // returns undefined

One solution would be to merge the custom input formats with the defaults. This would fix the majority of cases. Ideally, the responsibility for default date formats would be delegated to momentjs and the DateParser wouldn't attempt to "standardise" the separators.

**Comment by dchambers** [comment]
@charliehw, might it make sense to have separate `DateParser` and `LocalizedDateParser` classes, where the `LocalizedDateParser` would not allow an input format to be specified, or would only allow input formats like `SHORT` and `LONG`? Or, or is the idea of providing localized date parsing just a bad one?

**Comment by charliehw** [comment]
@dchambers, I think localised date parsing is necessary in many cases. It's also common to have to parse in multiple formats, the use case above with only one input format is just heavily simplified. So more often than not, one would have to add both a DateParser and a LocalizedDateParser to a property (does it make sense to add more than one parser to a property? is this possible?).

I think a LocalizedDateParser wouldn't need to have the input format specified at all. It would attempt to parse using all standard locale-specific date formats (e.g. those defined in momentjs).

**Comment by charliehw** [comment]
It just seems that DateParser is doing the bare minimum

**Comment by dchambers** [comment]
I've just taken a look and Moment.js already seems to support parsing localized date formats. For example, if you execute:

```js
moment('13/12/2000', 'l').format('DD/MM/YYYY');
```

You will get `Invalid date`, but if you first execute:

```js
moment.locale('en-GB');
```

then you will be able to parse `13/12/2000` with no problems.

There are lots more locale specific formatting codes available in addition to `l` &mdash; full details available within the Moment.js [Format](http://momentjs.com/docs/#/displaying/format/) section.

This would seem to suggest that we don't need to bother supporting locale specific parsing at all, and should leave it all to Moment.js.

**Comment by andyberry88** [comment]
@dchambers is this something that has been fixed and should it be in the 0.15.2 release?

**Comment by dchambers** [comment]
@andyberry88, yes, this was fixed in #1275.
